File: hto/15/hto_15.cpp
#include <iostream>
#include <bitset>
using namespace std;

class Solution {
public:
    int hammingWeight(uint32_t n) {
        return bitset<32>(n).count();
    }
};

int main() {
    Solution s;
    cout << s.hammingWeight(1) << endl;
}

File: hto/15/hto_15.go
func hammingWeight(num uint32) int {
    count := 0
    var flag uint32 = 1
    for flag > 0 {
        if num & flag > 0 {
            count++
        }
        flag <<= 1
    }
    return count
} 

func hammingWeight(num uint32) int {
    count := 0
    for num > 0 {
        count++
        num &= (num-1)
    }
    return count
}
